idd/schema/idd_parser.py

```diff
@@ -288,20 +337,46 @@ def parse_obj(data):
         elif token == TOKEN_OBSOLETE:
             next_token(data)
             if look_ahead(data) != TOKEN_STRING:
-                raise RuntimeError("expected string after /obsolete")
+                raise IddParsingError(
+                    message="In object '{obj_name}', expected string after /obsolete".format(
+                        obj_name=obj_name,
+                    ), data=data,
+                )
             if 'obsolete' in root:
-                raise RuntimeError("cannot have duplicate obsolete")
+                raise IddParsingError("cannot have duplicate obsolete")
```

Comment:
I haven't modifier ALL of the errors in the file to use the new helpers but I did a few. I don't think a lot of them are really common, and it's not that hard to track these down.

Reply:
:+1: